src/node_api.cc

```diff
@@ -705,6 +706,27 @@ bool FindWrapper(v8::Local<v8::Object> obj,
   return true;
 }
 
+std::map<v8::Isolate*, napi_env> napi_env_map;
+uv_once_t napi_env_once = UV_ONCE_INIT;
+uv_mutex_t napi_env_mutex;
```

Comment:
`node::Mutex` (and use `node::Mutex::ScopedLock` below), then you won't need a uv_once_t either.